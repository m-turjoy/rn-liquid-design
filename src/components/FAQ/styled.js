import styled from 'styled-components';
import {
  fontFamily,
  fontSize,
  fontWeight,
  color,
  lineHeight,
  width,
} from 'styled-system';

const DescriptionWrapper = styled.Text`
  ${fontSize}
  ${fontFamily}
${fontWeight}
${lineHeight}
${color}
${width}
padding-horizontal: 35px;
  text-align: center;
  padding-bottom: 50px;
  padding-top: 20px;
`;

const FAQWrapper = styled.View`
  ${width}
  align-items: center;
  justify-content: center;
`;

export { DescriptionWrapper, FAQWrapper };
