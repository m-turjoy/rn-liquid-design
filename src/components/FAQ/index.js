import React from 'react';
import {
  array,
  arrayOf,
  bool,
  oneOf,
  oneOfType,
  number,
  string,
  func,
} from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { theme, fonts, colors } from '../../config';
import Headline from '../Headline';
import Accordion from '../Accordion';
import { DescriptionWrapper, FAQWrapper } from './styled';
import { defaultThemeName } from '../../config/theme';

const FAQ = ({
  headlineText,
  headlineType,
  headlineColor,
  description,
  activeSections,
  expandMultiple,
  onChange,
  sections,
  duration,
  initiallyActiveSection,
  activeSection,
  disabled,
  onAnimationEnd,
  width,
  borderColor,
  titlefontFamily,
  titleFontSize,
  titleLineHeight,
  titleFontWeight,
  iconColor,
  iconSize,
  contentFontFamily,
  contentColor,
  inactiveTitleColor,
  activeTitleColor,
  contentFontSize,
  contentLineHeight,
  contentFontWeight,
  contentWidth,
  contentHeight,
  descriptionFontFamily,
  descriptionFontSize,
  descriptionFontWeight,
  descriptionLineHeight,
  descriptionColor,
  descriptionPadding,
  moduleWidth,
  themeName,
}) => (
  <ThemeProvider theme={theme.themes[themeName]}>
    <FAQWrapper width={moduleWidth}>
      <Headline type={headlineType} text={headlineText} color={headlineColor} />
      <DescriptionWrapper
        fontFamily={descriptionFontFamily}
        fontSize={descriptionFontSize}
        fontWeight={descriptionFontWeight}
        color={descriptionColor}
        lineHeight={descriptionLineHeight}
        paddingHorizontal={descriptionPadding}
        width={width}
      >
        {description}
      </DescriptionWrapper>
      <Accordion
        activeSections={activeSections}
        expandMultiple={expandMultiple}
        onChange={onChange}
        sections={sections}
        duration={duration}
        initiallyActiveSection={initiallyActiveSection}
        activeSection={activeSection}
        disabled={disabled}
        onAnimationEnd={onAnimationEnd}
        width={width}
        borderColor={borderColor}
        titlefontFamily={titlefontFamily}
        titleFontSize={titleFontSize}
        titleLineHeight={titleLineHeight}
        titleFontWeight={titleFontWeight}
        iconColor={iconColor}
        iconSize={iconSize}
        contentFontFamily={contentFontFamily}
        contentColor={contentColor}
        inactiveTitleColor={inactiveTitleColor}
        activeTitleColor={activeTitleColor}
        contentFontSize={contentFontSize}
        contentLineHeight={contentLineHeight}
        contentFontWeight={contentFontWeight}
        contentWidth={contentWidth}
        contentHeight={contentHeight}
        themeName={themeName}
      />
    </FAQWrapper>
  </ThemeProvider>
);

FAQ.propTypes = {
  headlineType: string,
  headlineText: string,
  headlineColor: string,
  description: string,
  sections: array.isRequired,
  onChange: func,
  duration: number,
  initiallyActiveSection: number,
  activeSection: oneOfType([bool, number]),
  disabled: bool,
  onAnimationEnd: func,
  width: number,
  borderColor: string,
  titlefontFamily: string,
  titleFontSize: number,
  titleLineHeight: number,
  titleFontWeight: number,
  iconColor: string,
  iconSize: number,
  contentFontFamily: string,
  contentColor: string,
  inactiveTitleColor: string,
  activeTitleColor: string,
  contentFontSize: number,
  contentLineHeight: number,
  contentFontWeight: number,
  contentWidth: string,
  contentHeight: number,
  expandMultiple: bool,
  activeSections: oneOf(arrayOf(string)),
  descriptionFontFamily: string,
  descriptionFontSize: number,
  descriptionFontWeight: number,
  descriptionLineHeight: number,
  descriptionColor: string,
  descriptionPadding: number,
  moduleWidth: string,
  themeName: string,
};

FAQ.defaultProps = {
  headlineType: 'H3',
  headlineText: 'Headline Insert',
  headlineColor: colors.richBlackDefault,
  description:
    'We have been around for 350 years, yet our majority owners are still the descendants of Friedrich Jacob Merck',
  descriptionFontFamily: fonts.Regular,
  descriptionFontSize: 14,
  descriptionLineHeight: 24.5,
  descriptionColor: colors.richBlackLightest,
  moduleWidth: '100%',
  themeName: defaultThemeName,
  width: 335,
};

export default FAQ;
