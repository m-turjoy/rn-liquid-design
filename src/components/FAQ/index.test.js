import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import FAQ from './index'

const SECTIONS = [
  {
    title: 'Section Title',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elita.'
  },
  {
    title: 'Section Title',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing.'
  }
]

configure({ adapter: new Adapter() })
describe('FAQ component', () => {
  describe('Renders correctly', () => {
    test('it renders default FAQ', () => {
      const tree = shallow(
        <FAQ
          sections={SECTIONS}
        />,
      )
      expect(tree).toMatchSnapshot()
    })
  })
})
