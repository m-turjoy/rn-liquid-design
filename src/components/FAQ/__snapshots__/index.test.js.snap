// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ component Renders correctly it renders default FAQ 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FAQ
    description="We have been around for 350 years, yet our majority owners are still the descendants of Friedrich Jacob Merck"
    descriptionColor="#93939d"
    descriptionFontFamily="Lato-Regular"
    descriptionFontSize={14}
    descriptionLineHeight={24.5}
    headlineColor="#1b1b25"
    headlineText="Headline Insert"
    headlineType="H3"
    moduleWidth="100%"
    sections={
      Array [
        Object {
          "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elita.",
          "title": "Section Title",
        },
        Object {
          "content": "Lorem ipsum dolor sit amet, consectetur adipiscing.",
          "title": "Section Title",
        },
      ]
    }
    themeName="vibrantCyan"
    width={335}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(View)
        width="100%"
      >
        <Headline
          color="#1b1b25"
          fontFamily="Lato-Black"
          fontSize={14}
          lineHeight={14}
          text="Headline Insert"
          textStyle={Object {}}
          type="H3"
        />
        <Styled(Text)
          color="#93939d"
          fontFamily="Lato-Regular"
          fontSize={14}
          lineHeight={24.5}
          width={335}
        >
          We have been around for 350 years, yet our majority owners are still the descendants of Friedrich Jacob Merck
        </Styled(Text)>
        <Accordion
          borderColor="#efeff1"
          contentColor="#1b1b25"
          contentFontFamily="Lato-Regular"
          contentFontSize={14}
          contentHeight={125}
          contentLineHeight={24.5}
          contentWidth="100%"
          disabled={false}
          duration={200}
          inactiveTitleColor="#1b1b25"
          onAnimationEnd={[Function]}
          sections={
            Array [
              Object {
                "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elita.",
                "title": "Section Title",
              },
              Object {
                "content": "Lorem ipsum dolor sit amet, consectetur adipiscing.",
                "title": "Section Title",
              },
            ]
          }
          themeName="vibrantCyan"
          titleFontFamily="Lato-Black"
          titleFontSize={16}
          titleLineHeight={20}
          width={335}
        />
      </Styled(View)>,
      "theme": Object {
        "colors": Object {
          "black": Object {
            "base": undefined,
          },
          "primary": Object {
            "base": "#2dbecd",
            "dark": "#29abb9",
            "darker": "#208590",
            "light": "#57cbd7",
            "lightest": "#abe5eb",
          },
          "richBlack": Object {
            "base": undefined,
            "dark": undefined,
            "darker": undefined,
            "darkest": undefined,
            "light": undefined,
            "lighter": undefined,
            "lightest": undefined,
          },
          "richBlue": Object {
            "base": "#0f69af",
            "dark": "#0e5f9e",
            "darker": "#0b4a7b",
            "darkest": "#083558",
            "light": "#3f87bf",
            "lighter": "#6fa5cf",
            "lightest": "#9fc3df",
          },
          "richGreen": Object {
            "base": "#149b5f",
            "dark": "#0d6c42",
            "darker": "#0c5d39",
            "darkest": "#094d2f",
            "light": "#5ab98f",
            "lighter": "#72c39f",
            "lightest": "#89cdaf",
          },
          "richPurple": Object {
            "base": "#503291",
            "dark": "#482d83",
            "darker": "#382366",
            "darkest": "#281949",
            "light": "#735ba7",
            "lighter": "#9684bd",
            "lightest": "#b9add3",
          },
          "richRed": Object {
            "base": "#e61e50",
            "dark": "#cf1b48",
            "darker": "#a11538",
            "darkest": "#730f28",
            "light": "#eb4b73",
            "lighter": "#f07896",
            "lightest": "#f5a5b9",
          },
          "secondary": Object {
            "base": "#ffc832",
            "dark": "#f3be2f",
            "darker": "#ddac28",
            "light": "#ffd35b",
            "lightest": "#ffe9ad",
          },
          "sensitiveGrey": Object {
            "base": "#f8f8fc",
            "dark": "#f3f3f7",
            "darker": "#efeff1",
            "darkest": "#e9e9ed",
            "light": "#f8f8fc",
            "lighter": "#ffffff",
            "lightest": "#ffffff",
          },
          "vibrantCyan": Object {
            "base": "#2dbecd",
            "dark": "#29abb9",
            "darker": "#208590",
            "darkest": "#175f67",
            "light": "#57cbd7",
            "lighter": "#81d8e1",
            "lightest": "#abe5eb",
          },
          "vibrantGreen": Object {
            "base": undefined,
            "dark": undefined,
            "darker": undefined,
            "darkest": undefined,
            "light": undefined,
            "lighter": undefined,
            "lightest": undefined,
          },
          "vibrantMagenta": Object {
            "base": "#eb3c96",
            "dark": "#d43687",
            "darker": "#a52a69",
            "darkest": "#761e4b",
            "light": "#ef63ab",
            "lighter": "#f38ac0",
            "lightest": "#f7b1d5",
          },
          "vibrantYellow": Object {
            "base": "#ffc832",
            "dark": "#f3be2f",
            "darker": "#ddac28",
            "darkest": "#806419",
            "light": "#ffd35b",
            "lighter": "#ffde84",
            "lightest": "#ffe9ad",
          },
          "white": Object {
            "base": "#ffffff",
          },
        },
        "fontSizes": Array [
          8,
          10,
          12,
          14,
          16,
          18,
          20,
          24,
          36,
        ],
        "fontWeights": Array [
          100,
          200,
          300,
          400,
          500,
          600,
          700,
          800,
          900,
        ],
        "fonts": Object {
          "Black": "Lato-Black",
          "Bold": "Lato-Bold",
          "IcoMoon": "icomoon",
          "Merck": "MerckWeb-Regular",
          "Regular": "Lato-Regular",
        },
        "heights": Object {
          "Big": 50,
          "Default": 40,
        },
        "shadows": Object {
          "active": Object {
            "elevation": 40,
            "shadowColor": "#1b1b25",
            "shadowOffset": "0px 30px",
            "shadowOpacity": 0.3,
            "shadowRadius": 20,
          },
          "default": Object {},
          "stacked": Object {
            "elevation": 25,
            "shadowColor": "#1b1b25",
            "shadowOffset": "0px 30px",
            "shadowOpacity": 0.3,
            "shadowRadius": 20,
          },
        },
        "space": Array [
          0,
          1,
          2,
          5,
          10,
          15,
          20,
          24,
          50,
          8,
          7,
          4,
          9,
          6,
        ],
        "stacked": Object {
          "stackedFirst": Object {
            "borderBottomWidth": 1,
            "borderColor": "#f3f3f7",
            "borderRightWidth": 1,
            "position": "absolute",
            "zIndex": 10,
          },
          "stackedSecond": Object {
            "borderBottomWidth": 1,
            "borderColor": "#f3f3f7",
            "borderRightWidth": 1,
            "left": 3,
            "position": "absolute",
            "top": 3,
            "zIndex": 9,
          },
          "stackedThird": Object {
            "borderBottomWidth": 1,
            "borderColor": "#f3f3f7",
            "borderRightWidth": 1,
            "left": 6,
            "top": 6,
            "zIndex": 8,
          },
        },
        "widths": Array [
          50,
          100,
          130,
        ],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Headline
            color="#1b1b25"
            fontFamily="Lato-Black"
            fontSize={14}
            lineHeight={14}
            text="Headline Insert"
            textStyle={Object {}}
            type="H3"
          />,
          <Styled(Text)
            color="#93939d"
            fontFamily="Lato-Regular"
            fontSize={14}
            lineHeight={24.5}
            width={335}
          >
            We have been around for 350 years, yet our majority owners are still the descendants of Friedrich Jacob Merck
          </Styled(Text)>,
          <Accordion
            borderColor="#efeff1"
            contentColor="#1b1b25"
            contentFontFamily="Lato-Regular"
            contentFontSize={14}
            contentHeight={125}
            contentLineHeight={24.5}
            contentWidth="100%"
            disabled={false}
            duration={200}
            inactiveTitleColor="#1b1b25"
            onAnimationEnd={[Function]}
            sections={
              Array [
                Object {
                  "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elita.",
                  "title": "Section Title",
                },
                Object {
                  "content": "Lorem ipsum dolor sit amet, consectetur adipiscing.",
                  "title": "Section Title",
                },
              ]
            }
            themeName="vibrantCyan"
            titleFontFamily="Lato-Black"
            titleFontSize={16}
            titleLineHeight={20}
            width={335}
          />,
        ],
        "width": "100%",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#1b1b25",
            "fontFamily": "Lato-Black",
            "fontSize": 14,
            "lineHeight": 14,
            "text": "Headline Insert",
            "textStyle": Object {},
            "type": "H3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "We have been around for 350 years, yet our majority owners are still the descendants of Friedrich Jacob Merck",
            "color": "#93939d",
            "fontFamily": "Lato-Regular",
            "fontSize": 14,
            "fontWeight": undefined,
            "lineHeight": 24.5,
            "paddingHorizontal": undefined,
            "width": 335,
          },
          "ref": null,
          "rendered": "We have been around for 350 years, yet our majority owners are still the descendants of Friedrich Jacob Merck",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeSection": undefined,
            "activeSections": undefined,
            "activeTitleColor": undefined,
            "borderColor": "#efeff1",
            "contentColor": "#1b1b25",
            "contentFontFamily": "Lato-Regular",
            "contentFontSize": 14,
            "contentFontWeight": undefined,
            "contentHeight": 125,
            "contentLineHeight": 24.5,
            "contentWidth": "100%",
            "disabled": false,
            "duration": 200,
            "expandMultiple": undefined,
            "iconColor": undefined,
            "iconSize": undefined,
            "inactiveTitleColor": "#1b1b25",
            "initiallyActiveSection": undefined,
            "onAnimationEnd": [Function],
            "onChange": undefined,
            "sections": Array [
              Object {
                "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elita.",
                "title": "Section Title",
              },
              Object {
                "content": "Lorem ipsum dolor sit amet, consectetur adipiscing.",
                "title": "Section Title",
              },
            ],
            "themeName": "vibrantCyan",
            "titleFontFamily": "Lato-Black",
            "titleFontSize": 16,
            "titleFontWeight": undefined,
            "titleLineHeight": 20,
            "titlefontFamily": undefined,
            "width": 335,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(View)
          width="100%"
        >
          <Headline
            color="#1b1b25"
            fontFamily="Lato-Black"
            fontSize={14}
            lineHeight={14}
            text="Headline Insert"
            textStyle={Object {}}
            type="H3"
          />
          <Styled(Text)
            color="#93939d"
            fontFamily="Lato-Regular"
            fontSize={14}
            lineHeight={24.5}
            width={335}
          >
            We have been around for 350 years, yet our majority owners are still the descendants of Friedrich Jacob Merck
          </Styled(Text)>
          <Accordion
            borderColor="#efeff1"
            contentColor="#1b1b25"
            contentFontFamily="Lato-Regular"
            contentFontSize={14}
            contentHeight={125}
            contentLineHeight={24.5}
            contentWidth="100%"
            disabled={false}
            duration={200}
            inactiveTitleColor="#1b1b25"
            onAnimationEnd={[Function]}
            sections={
              Array [
                Object {
                  "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elita.",
                  "title": "Section Title",
                },
                Object {
                  "content": "Lorem ipsum dolor sit amet, consectetur adipiscing.",
                  "title": "Section Title",
                },
              ]
            }
            themeName="vibrantCyan"
            titleFontFamily="Lato-Black"
            titleFontSize={16}
            titleLineHeight={20}
            width={335}
          />
        </Styled(View)>,
        "theme": Object {
          "colors": Object {
            "black": Object {
              "base": undefined,
            },
            "primary": Object {
              "base": "#2dbecd",
              "dark": "#29abb9",
              "darker": "#208590",
              "light": "#57cbd7",
              "lightest": "#abe5eb",
            },
            "richBlack": Object {
              "base": undefined,
              "dark": undefined,
              "darker": undefined,
              "darkest": undefined,
              "light": undefined,
              "lighter": undefined,
              "lightest": undefined,
            },
            "richBlue": Object {
              "base": "#0f69af",
              "dark": "#0e5f9e",
              "darker": "#0b4a7b",
              "darkest": "#083558",
              "light": "#3f87bf",
              "lighter": "#6fa5cf",
              "lightest": "#9fc3df",
            },
            "richGreen": Object {
              "base": "#149b5f",
              "dark": "#0d6c42",
              "darker": "#0c5d39",
              "darkest": "#094d2f",
              "light": "#5ab98f",
              "lighter": "#72c39f",
              "lightest": "#89cdaf",
            },
            "richPurple": Object {
              "base": "#503291",
              "dark": "#482d83",
              "darker": "#382366",
              "darkest": "#281949",
              "light": "#735ba7",
              "lighter": "#9684bd",
              "lightest": "#b9add3",
            },
            "richRed": Object {
              "base": "#e61e50",
              "dark": "#cf1b48",
              "darker": "#a11538",
              "darkest": "#730f28",
              "light": "#eb4b73",
              "lighter": "#f07896",
              "lightest": "#f5a5b9",
            },
            "secondary": Object {
              "base": "#ffc832",
              "dark": "#f3be2f",
              "darker": "#ddac28",
              "light": "#ffd35b",
              "lightest": "#ffe9ad",
            },
            "sensitiveGrey": Object {
              "base": "#f8f8fc",
              "dark": "#f3f3f7",
              "darker": "#efeff1",
              "darkest": "#e9e9ed",
              "light": "#f8f8fc",
              "lighter": "#ffffff",
              "lightest": "#ffffff",
            },
            "vibrantCyan": Object {
              "base": "#2dbecd",
              "dark": "#29abb9",
              "darker": "#208590",
              "darkest": "#175f67",
              "light": "#57cbd7",
              "lighter": "#81d8e1",
              "lightest": "#abe5eb",
            },
            "vibrantGreen": Object {
              "base": undefined,
              "dark": undefined,
              "darker": undefined,
              "darkest": undefined,
              "light": undefined,
              "lighter": undefined,
              "lightest": undefined,
            },
            "vibrantMagenta": Object {
              "base": "#eb3c96",
              "dark": "#d43687",
              "darker": "#a52a69",
              "darkest": "#761e4b",
              "light": "#ef63ab",
              "lighter": "#f38ac0",
              "lightest": "#f7b1d5",
            },
            "vibrantYellow": Object {
              "base": "#ffc832",
              "dark": "#f3be2f",
              "darker": "#ddac28",
              "darkest": "#806419",
              "light": "#ffd35b",
              "lighter": "#ffde84",
              "lightest": "#ffe9ad",
            },
            "white": Object {
              "base": "#ffffff",
            },
          },
          "fontSizes": Array [
            8,
            10,
            12,
            14,
            16,
            18,
            20,
            24,
            36,
          ],
          "fontWeights": Array [
            100,
            200,
            300,
            400,
            500,
            600,
            700,
            800,
            900,
          ],
          "fonts": Object {
            "Black": "Lato-Black",
            "Bold": "Lato-Bold",
            "IcoMoon": "icomoon",
            "Merck": "MerckWeb-Regular",
            "Regular": "Lato-Regular",
          },
          "heights": Object {
            "Big": 50,
            "Default": 40,
          },
          "shadows": Object {
            "active": Object {
              "elevation": 40,
              "shadowColor": "#1b1b25",
              "shadowOffset": "0px 30px",
              "shadowOpacity": 0.3,
              "shadowRadius": 20,
            },
            "default": Object {},
            "stacked": Object {
              "elevation": 25,
              "shadowColor": "#1b1b25",
              "shadowOffset": "0px 30px",
              "shadowOpacity": 0.3,
              "shadowRadius": 20,
            },
          },
          "space": Array [
            0,
            1,
            2,
            5,
            10,
            15,
            20,
            24,
            50,
            8,
            7,
            4,
            9,
            6,
          ],
          "stacked": Object {
            "stackedFirst": Object {
              "borderBottomWidth": 1,
              "borderColor": "#f3f3f7",
              "borderRightWidth": 1,
              "position": "absolute",
              "zIndex": 10,
            },
            "stackedSecond": Object {
              "borderBottomWidth": 1,
              "borderColor": "#f3f3f7",
              "borderRightWidth": 1,
              "left": 3,
              "position": "absolute",
              "top": 3,
              "zIndex": 9,
            },
            "stackedThird": Object {
              "borderBottomWidth": 1,
              "borderColor": "#f3f3f7",
              "borderRightWidth": 1,
              "left": 6,
              "top": 6,
              "zIndex": 8,
            },
          },
          "widths": Array [
            50,
            100,
            130,
          ],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Headline
              color="#1b1b25"
              fontFamily="Lato-Black"
              fontSize={14}
              lineHeight={14}
              text="Headline Insert"
              textStyle={Object {}}
              type="H3"
            />,
            <Styled(Text)
              color="#93939d"
              fontFamily="Lato-Regular"
              fontSize={14}
              lineHeight={24.5}
              width={335}
            >
              We have been around for 350 years, yet our majority owners are still the descendants of Friedrich Jacob Merck
            </Styled(Text)>,
            <Accordion
              borderColor="#efeff1"
              contentColor="#1b1b25"
              contentFontFamily="Lato-Regular"
              contentFontSize={14}
              contentHeight={125}
              contentLineHeight={24.5}
              contentWidth="100%"
              disabled={false}
              duration={200}
              inactiveTitleColor="#1b1b25"
              onAnimationEnd={[Function]}
              sections={
                Array [
                  Object {
                    "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elita.",
                    "title": "Section Title",
                  },
                  Object {
                    "content": "Lorem ipsum dolor sit amet, consectetur adipiscing.",
                    "title": "Section Title",
                  },
                ]
              }
              themeName="vibrantCyan"
              titleFontFamily="Lato-Black"
              titleFontSize={16}
              titleLineHeight={20}
              width={335}
            />,
          ],
          "width": "100%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "#1b1b25",
              "fontFamily": "Lato-Black",
              "fontSize": 14,
              "lineHeight": 14,
              "text": "Headline Insert",
              "textStyle": Object {},
              "type": "H3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "We have been around for 350 years, yet our majority owners are still the descendants of Friedrich Jacob Merck",
              "color": "#93939d",
              "fontFamily": "Lato-Regular",
              "fontSize": 14,
              "fontWeight": undefined,
              "lineHeight": 24.5,
              "paddingHorizontal": undefined,
              "width": 335,
            },
            "ref": null,
            "rendered": "We have been around for 350 years, yet our majority owners are still the descendants of Friedrich Jacob Merck",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeSection": undefined,
              "activeSections": undefined,
              "activeTitleColor": undefined,
              "borderColor": "#efeff1",
              "contentColor": "#1b1b25",
              "contentFontFamily": "Lato-Regular",
              "contentFontSize": 14,
              "contentFontWeight": undefined,
              "contentHeight": 125,
              "contentLineHeight": 24.5,
              "contentWidth": "100%",
              "disabled": false,
              "duration": 200,
              "expandMultiple": undefined,
              "iconColor": undefined,
              "iconSize": undefined,
              "inactiveTitleColor": "#1b1b25",
              "initiallyActiveSection": undefined,
              "onAnimationEnd": [Function],
              "onChange": undefined,
              "sections": Array [
                Object {
                  "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elita.",
                  "title": "Section Title",
                },
                Object {
                  "content": "Lorem ipsum dolor sit amet, consectetur adipiscing.",
                  "title": "Section Title",
                },
              ],
              "themeName": "vibrantCyan",
              "titleFontFamily": "Lato-Black",
              "titleFontSize": 16,
              "titleFontWeight": undefined,
              "titleLineHeight": 20,
              "titlefontFamily": undefined,
              "width": 335,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
